### PR TITLE
feat(canary): add Pub/Sub handler

### LINKED ISSUE #2182: Add Pub/Sub handler  to canary-bot
Currently canary-bot has a normal github webhook hander and several cron handlers. It's great if there's Pub/Sub handler too.

### PR DESCRIPTION
**Pull request by tmatsuo**
also add corresponding code to serverless proxy

fixes #2182

### CODE REVIEW

File: packages/canary-bot/src/canary-bot.ts [hunk 1]
```diff
@@ -40,6 +42,14 @@ const cronIssueTitle = 'A canary is chirping';
 const myRepositoryName = 'repo-automation-bots';
 const myOrganizationName = 'googleapis';
 
+interface PubSubContext {
+  github: Octokit;
+  readonly event: string;
+  log: Logger;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  payload: any;
```
Comment (JustinBeckwith) [inline packages/canary-bot/src/canary-bot.ts:50]: Would it be possible to make this interface `<T = any>` so consumers can attach a type to it?

File: packages/canary-bot/src/canary-bot.ts [hunk 2]
```diff
@@ -117,4 +127,12 @@ export = (app: Probot) => {
       );
     }
   });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  app.on('pubsub.message' as any, async (context: PubSubContext) => {
```
Comment (JustinBeckwith) [inline packages/canary-bot/src/canary-bot.ts:132]: I've found you can cast to `as '*'` and get the same effect without the `any`.

File: serverless-scheduler-proxy/main.go [hunk 3]
```diff
@@ -117,6 +118,23 @@ func rewriteBotContainerURL(c botConfig) func(*http.Request) {
 	}
 }
 
+func rewriteBotCanaryURL(c botConfig) func(*http.Request) {
```
Comment (JustinBeckwith) [inline serverless-scheduler-proxy/main.go:121]: I had to stare at this code for longer than I'd care to admit before I realized I wasn't looking at JavaScript.

**Review by JustinBeckwith — APPROVED** [review]

File: packages/canary-bot/src/canary-bot.ts (hunk 1 above)
Comment (tmatsuo) [reply packages/canary-bot/src/canary-bot.ts:50]: Sure

File: packages/canary-bot/src/canary-bot.ts (hunk 2 above)
Comment (tmatsuo) [reply packages/canary-bot/src/canary-bot.ts:132]: Unfortunately we can not. flakybot and owl-bot both use `any` for the pubsub handler.

**Pull request merged**